modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -517,6 +522,10 @@ private CompletableFuture<?> processRequest(ReplicaRequest request, @Nullable Bo
                         processOperationRequestWithTxRwCounter(senderId, request, isPrimary, opTsIfDirectRo, leaseStartTime));
     }
 
+    private CompletableFuture<Long> processGetEstimatedSizeRequest() {
+        return completedFuture(mvDataStorage.estimatedSize());
```

Comment:
Curious whether estimatedSize evaluates writeIntent as +1?

Reply:
No, Write Intents are going to be omitted, only committed entries change the size

Reply:
Sounds reasonable while we retrieving data from primary.